packages/babel-plugin-proposal-decorators/src/transformer-2021-12.ts

```diff
@@ -967,8 +964,8 @@ function transformClass(
             t.arrayPattern(locals),
             t.callExpression(state.addHelper("applyDecs"), [
               t.thisExpression(),
-              elementDecorations,
               classDecorations,
+              elementDecorations,
```

Comment:
This ensures that class decorator expressions are evaluated before member decorator expressions.